lib/JSHandle.js

```diff
@@ -234,6 +238,19 @@ class ElementHandle extends JSHandle {
     ];
   }
 
+  /**
+   * @param {!Array<{x: number, y: number}>} quad
+   * @param {number} width
+   * @param {number} height
+   * @return {!Array<{x: number, y: number}>}
+   */
+  _intersectQuadWithViewport(quad, width, height) {
+    return quad.map(point => ({
+      x: Math.min(Math.max(point.x, 0), width),
```

Comment:
Nobody has added clamp to JavaScript yet? There’s probably an npm package for it :)